### ISSUE TITLE
AKS adds/uses deprecated failure-domain.beta.kubernetes.io labels

### ISSUE THREAD

**Issue by antonmatsiuk** [Azure/AKS#2188]
**What happened**:
AKS uses depreacted `failure-domain.beta.kubernetes.io` labels for availability zones 

**What you expected to happen**:

According to this [doc](https://kubernetes.io/docs/reference/labels-annotations-taints/)
`failure-domain.beta.kubernetes.io/region` and `failure-domain.beta.kubernetes.io/zone` labels are deprecated in favor
of `topology.kubernetes.io/region` and `topology.kubernetes.io/zone` labels starting from v.1.17. AKS still uses this labels while creating dynamic PVs and also reacts on them in statically created PVs and ignores new `topology.kubernetes.io` labels. 

**How to reproduce it (as minimally and precisely as possible)**:
1. create a test deployment with static PV and new labels:
```
topology.kubernetes.io/region: 'germanywestcentral'
topology.kubernetes.io/zone: 'germanywestcentral-3'
```
following error appears: 
```
HTTPStatusCode: 400, RawError: { "error": { "code": "BadRequest", "message": "Disk /subscriptions/4782a679-70e8-4367-b5a6-ba4fd408973a/resourceGroups/FCLD-AKS-storage/providers/Microsoft.Compute/disks/pv-grafana cannot be attached to the VM because it is not in the same zone as the VM. VM zone: '2'. Disk zone: '3'." } }
```
2. create a test deployment with static PV and deprecated labels:
```
 failure-domain.beta.kubernetes.io/region: 'germanywestcentral'
 failure-domain.beta.kubernetes.io/zone: 'germanywestcentral-3'
```
Pod gets scheduled into a proper AZ

3. Create a deployment with dynamic PVC/PV. PV gets deprecated labels assigned. 
 
**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v.1.18.14
- AKS Location: Germany West Central

**Comment by andyzhangx** [comment]
we are still evaluating whether to change in-tree azure disk driver to use GA labels, there is a PR here: https://github.com/kubernetes/kubernetes/pull/99424#issuecomment-798905281

**Comment by andyzhangx** [comment]
already fixed in k8s 1.22
